tests/test_recursive.py

```diff
@@ -0,0 +1,253 @@
+        captured = capsys.readouterr()
+        assert "the following arguments are required: --sub-a" in captured.err
+
+    def test_flag(self):
+        config = parse(self.Config, ["mode", "--sub-a", "1", "--flag"])
+        assert config.mode == "mode"
+        assert config.sub.a == 1
+        assert config.sub.b == "abc"
+        assert config.flag is True
+
+    def test_set_sub(self):
+        config = parse(self.Config, ["mode", "--sub-a", "1", "--sub-b", "def"])
+        assert config.mode == "mode"
+        assert config.sub.a == 1
+        assert config.sub.b == "def"
+        assert config.flag is False
+
+
+class TestParseSubConfigWithPositional:
+    @dataclass
+    class Config:
+        mode: str = field(metadata=dict(positional=True))
+        sub: SubConfigWithPositional = field(default_factory=SubConfigWithPositional)
+        flag: bool = field(default=False, metadata=dict(as_flags=True))
+
+    def test_help(self, capsys):
+        with raises(SystemExit):
+            parse(self.Config, ["--help"], prog="prog")  # type: ignore
+        captured = capsys.readouterr()
+        assert "mode [sub_a]" in captured.out
+
+    def test_parse(self):
+        config = parse(self.Config, ["mode"])
+        assert config.mode == "mode"
+        assert config.sub.a == 42
+        assert config.sub.b == "abc"
+        assert config.flag is False
+
+    def test_flag(self):
+        config = parse(self.Config, ["mode", "1", "--flag"])
+        assert config.mode == "mode"
+        assert config.sub.a == 1
+        assert config.sub.b == "abc"
+        assert config.flag is True
+
+    def test_set_sub(self):
+        config = parse(
+            self.Config,
+            ["mode", "1", "--sub-b", "def"],
+        )
+        assert config.mode == "mode"
+        assert config.sub.a == 1
+        assert config.sub.b == "def"
+        assert config.flag is False
+
+
+class TestParsePositionalSubConfig:
+    @dataclass
+    class Config:
+        mode: str = field(metadata=dict(positional=True))
+        sub: SubConfig = field(default_factory=SubConfig, metadata=dict(positional=True))
+        flag: bool = field(default=False, metadata=dict(as_flags=True))
+
+    def test_parse(self):
+        with raises(ValueError):
+            parse(self.Config, ["mode"])
+
+
+class TestParseNestedSubConfig:
+    @dataclass
+    class Config:
+        mode: str = field(metadata=dict(positional=True))
+        sub: NestedSub = field(default_factory=NestedSub)
+        flag: bool = field(default=False, metadata=dict(as_flags=True))
+
+    @mark.parametrize(
+        "help_string",
+        [
+            "usage: prog [-h] [--sub-a SUB_A] [--sub-b-a SUB_B_A]",
+            "sub_b:",
+            "--sub-b-b SUB_B_B  a string (default: abc)\n",
+        ],
+    )
+    def test_help(self, capsys, help_string: str):
+        with raises(SystemExit):
+            parse(self.Config, ["--help"], prog="prog")  # type: ignore
```

Comment:
why does this raise a system exit?

Reply:
When calling `--help`, argparse always prints a help message and then exits.